### DISCUSSION TITLE
Functions to verify output and not to verify

### DISCUSSION THREAD

**Discussion by FxmousKid** [FxmousKid/42Cursus-minishell#6] [General]
As programmers, we have to do our due diligence making sure that we code is what we want to code, and to make sure of that, we can check if a certain function that returns its success status did, in fact, return success or failure, and treat it accordingly.

However, this habit can quickly delve into paranoia, indeed the line between overly cautious and careless is quite thin, but the goal of this discussion is to discuss how to reach it ; by deciding logically on which function poses fundamental risk of failing

For example : the printf function, can rarely fail, unless null is passed as argument. yes, the libc is not paranoiac and checks if every syscalls fails, since it's pretty rare.

But what does it even mean to handle a function that failed ? Well we can primarly do 2 different approaches : 
1) if the error is problematic
-  check for return value if it's failure
-  Alert the user if the error is critical, but most of the time just perror("")
-  debug(DBG("Failed to ...")) to alert the dev in debug mode
-  Free any allocated memory & close any open system ressources but be careful about exiting because you might create zombie processes...

but this approach is very tedious, verbose, can create zombie processes, and can obscure code and take away from readability, after all, code is read more than written. 

So what to do then ? well that proves quite difficult to answer and find a middle ground.

Personnaly i suggest to use this approach only with syscalls that can lots of user-prone errors such as close, open (which we have to check anyway for ERRNO values), access (same thing here), ... 

Basically this list : 

| Syscalls to check |
|--------|
| open |
| close |
| read |
| access |
| fork |
| pipe |
| dup  / dup2 |

**Comment by FxmousKid** [comment]
@MonkePlusPlus kesten pense

**Comment by FxmousKid** [comment]
@MonkePlusPlus fdp repond

**Comment by MonkePlusPlus** [comment]
I totally agree with that logic. Good job it's very well explain !

**Reply by FxmousKid** [reply]
tu ma juste balancer le default feedback de correction

**Reply by MonkePlusPlus** [reply]
👍
